[14.0][FIX] fix multiple subscription request glitches

huguesdk opened the pull request:
fix multiple glitches related to subscription requests:

* fix subscription request state when creating one from a partner.
* fix partner matching by email or company register number.
* fix subscription request name for companies.
* refactor `.create()` method to handle both people and companies.
* add several tests.
* fix email templates for companies.

robinkeunen commented on cooperator/data/mail_template_data.xml:
```diff
@@ -271,7 +271,7 @@
 <div style="font-family: 'Lucica Grande', Ubuntu, Arial, Verdana, sans-serif; font-size: 12px; color: rgb(34, 34, 34); background-color: #FFF; ">
 
-    <p>Hello ${object.firstname},</p>
+    <p>Hello ${object.subscription_request_ids[-1].firstname},</p>
```
Why the [-1] ? The order of moves is `_order = 'date desc, name desc, id desc'` so you'll end up with the oldest request, is that what you want ? 
Also, why do you index here and not on line 32 ?

huguesdk replied:
good catch, thanks! fixed.

on line 32, it is the `account.move` model, not the `res.partner` model, so the subscription request is directly accessible.

robinkeunen replied:
Ok sorry, pas vu que c'était deux templates différents.

robinkeunen commented on cooperator/models/subscription_request.py:
```diff
@@ -85,55 +80,59 @@ def _check_share_available_to_user(self):
+        else:
+            partner = partner_model.get_cooperator_from_email(vals.get("email"))
+        if partner:
+            vals["partner_id"] = partner.id
```
Don't you want to return a partner in this case as well ?

And if nothing is found, isn't it better to return an empty `RecordSet` rather than an implicit `None`

huguesdk replied:
my bad, thanks!

robinkeunen commented on cooperator/tests/test_cooperator.py:
```diff
@@ -210,3 +209,385 @@ def test_compute_is_valid_iban_on_subscription_request(self):
         self.request.iban = False
         self.assertTrue(request.is_valid_iban)
+
+    def _get_dummy_subscription_requests_vals(self):
```
thx for this 👍

huguesdk commented:
@cathLemb or @polchampion: could you please validate this functionally? it is deployed on the test servers (on all dbs, and should work on a new db also, of course). all changes are listed in the description. note: it’s on version 14.0.

huguesdk committed:
[FIX] fix multiple subscription request glitches

fix multiple glitches related to subscription requests:

* fix subscription request state when creating one from a partner.
* fix partner matching by email or company register number.
* fix subscription request name for companies.
* refactor .create() method to handle both people and companies.
* add several tests.
* fix email templates for companies.
* update cooperator.pot, fr and fr_be translations: the only change is
  the way to access the first name from the email templates; the reason
  that this results in quite some changes in the files is that this
  change causes the strings to be ordered differently.

huguesdk commented:
/ocabot merge minor
